dockeradapter/dockeradapter.go

```diff
@@ -130,18 +141,91 @@ func ImageExists(ctx context.Context, client ImageInspector, imageName string) b
 	return true
 }
 
-// New returns a client satisfying the Client interface
+// New returns a client connected with the highest API version supported by both the Lumogon client
+// and the Docker runtime
 func New() (Client, error) {
-	concreteClient := new(concreteDockerClient)
-	logging.Debug("[Docker Adapter] Creating container runtime client: Docker")
-	dockerAPIClient, err := client.NewEnvClient()
+
+	host, _, _ := DockerEnvvars()
+
+	serverAPIVersion, _, err := ServerInfo(host)
 	if err != nil {
-		return nil, fmt.Errorf("[Docker Adapter] Unable to initialise container runtime type: Docker, error: %s", err)
+		logging.Debug("[Docker Adapter] Unable to determine Docker Server API version: %s", err.Error())
+		return nil, err
```

Comment:
I think providing the context of the error (ie. what the previous return did) is probably worthwhile. Looking at the ServerInfo implementation is can return a number of different low-level errors and tracking those down could be trickier without the context. Any reason to change the return?

Reply:
Will revert